docs/conceptual/More-about-how-ROCm-uses-PCIe-Atomics.rst

```diff
@@ -1,156 +1,80 @@
+* `Xilinx PCIe Ultrascale White paper <https://docs.xilinx.com/v/u/8OZSA2V1b1LLU2rRCDVGQw>`_
 
-  * Mellanox Bluefield SOC
-  * Cavium Thunder X2
+Working of PCIe 3.0 in ROCm
+-------------------------------
+Starting from PCIe 3.0, atomic operations can be requested, routed through, and completed by PCIe components. Routing and completion do not require software support. Component support for each can be identified by the Device Capabilities 2 (DevCap2) register. Upstream
+bridges need to have atomic operations routing enabled. If not enabled, the atomic operations will fail even if the 
+PCIe endpoint and PCIe I/O devices can perform atomic operations.
 
-In ROCm, we also take advantage of PCIe ID based ordering technology for P2P when the GPU
-originates two writes to two different targets:
+To enable atomic operations routing between multiple Root Ports, each Root Port must support atomic operations routing. This capability can be identified from the atomic operations routing supported bit in the DevCap2 register. If the bit has value of 1, routing is supported.
 
-* Write to another GPU memory
-* Write to system memory to indicate transfer complete
-
-They are routed off to different ends of the computer but we want to make sure the write to system
-memory to indicate transfer complete occurs AFTER P2P write to GPU has complete.
-
-BAR memory overview
-----------------------------------------------------------------------------------------------------
-On a Xeon E5 based system in the BIOS we can turn on above 4GB PCIe addressing, if so he need to set
-memory-mapped input/output (MMIO) base address (MMIOH base) and range (MMIO high size) in the BIOS.
-
-In the Supermicro system in the system bios you need to see the following
-
-  * Advanced->PCIe/PCI/PnP configuration-\> Above 4G Decoding = Enabled
-  * Advanced->PCIe/PCI/PnP Configuration-\>MMIOH Base = 512G
-  * Advanced->PCIe/PCI/PnP Configuration-\>MMIO High Size = 256G
-
-When we support Large Bar Capability there is a Large Bar VBIOS which also disable the IO bar.
-
-For GFX9 and Vega10 which have Physical Address up 44 bit and 48 bit Virtual address.
-
-  * BAR0-1 registers: 64bit, prefetchable, GPU memory. 8GB or 16GB depending on Vega10 SKU. Must
-    be placed < 2^44 to support P2P  	access from other Vega10.
-  * BAR2-3 registers: 64bit, prefetchable, Doorbell. Must be placed \< 2^44 to support P2P access from
-    other Vega10.
-  * BAR4 register: Optional, not a boot device.
-  * BAR5 register: 32bit, non-prefetchable, MMIO. Must be placed \< 4GB.
-
-Here is how our base address register (BAR) works on GFX 8 GPUs with 40 bit Physical Address Limit ::
-
-  11:00.0 Display controller: Advanced Micro Devices, Inc. [AMD/ATI] Fiji [Radeon R9 FURY / NANO
-  Series] (rev c1)
+If your system has a PCIe Express Switch, it must support atomic operations routing. Atomic
```

Comment:
Is this that if the system has PCIe Express then it will support atomic operations or that you have to configure it to support atomic operations? Needs a bit more clarity.

Reply:
Updated the sentence.